File: Swift/guess-number-higher-or-lower.swift
class Solution {
    func guessNumber(_ n: Int) -> Int {
        var left = 1
        var right = n

        while left <= right {
            let mid = left + (right - left) / 2
            switch guess(mid) {
            case 0:
                return mid
            case 1:
                right = mid - 1
            case -1:
                left = mid + 1
            default:
                break
            }
        }

        fatalError("no valid outputs")
    }
}

File: Python/guess-number-higher-or-lower.py
class Solution:
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        left = 1
        right = n

        while left <= right:
            mid = left + (right - left) // 2
            res = guess(mid)
            if res == 0:
                return mid
            elif res == 1:
                right = mid - 1
            elif res == -1:
                left = mid + 1
